File: algorithms/2_medium/C/03759_count_elements_with_at_least_k_greater_values/solution.py
class Solution:
    def countElements(self, nums: List[int], k: int) -> int:
        n = len(nums)
        if k == 0:
            return n
        nums.sort()
        k_th_biggest = nums[n - k]

        return bisect.bisect_left(nums, k_th_biggest)

File: algorithms/2_medium/C/03759_count_elements_with_at_least_k_greater_values/solution.cpp
class Solution {
public:
    static int countElements(vector<int>& nums, int k) {
        const size_t len{ nums.size() };

        if (k == 0) {
            return static_cast<int>(len);
        }

        ranges::sort(nums);
        const int k_th_biggest{ nums[len - static_cast<size_t>(k)] };

        return static_cast<int>(ranges::lower_bound(nums, k_th_biggest) - nums.begin());
    }
};
